.azure-pipelines/templates/code_quality_checks.yml

```diff
@@ -41,6 +41,18 @@ stages:
               find . -type f -name CMakeLists.txt -or -name '*.cmake' | xargs cmake-format --check
             displayName: 'Run cmake-format'
 
+      - job: 'copyright_year'
+        displayName: 'Copyright Year'
+        pool:
+          vmImage: 'ubuntu-latest'
+        steps:
```

Comment:
Is see this takes only 5 seconds, and I wonder how it can be so fast? Is it running in a new instance of a VM, or reusing a previously loaded one? Some magic Azure is doing?

Reply:
I think it only counts time spend on the job, not VM cycling time.

They likely have a large pool of VMs of each image ready to be used by incoming jobs, so startup time is minimal assuming the pool is healthy.
Once a VM has been used it is killed and more are spawned to keep the pool size up (likely from a suspended snapshot to keep spawn time down).

Reply:
As for the checkout, that is likely fast as the orchestration layer has already done the Git pull, etc. in order to actually read the pipeline scripts, therefore it is just IO within a single datacentre.